src/parser/lexer.mll

```diff
@@ -262,6 +262,8 @@ let raw_markup =
   ([^ '%'] | '%'+ [^ '%' '}'])* '%'*
 let raw_markup_target =
   ([^ ':' '%'] | '%'+ [^ ':' '%' '}'])* '%'*
+let code_block_meta =
+  ([^ '['])*
```

Comment:
This parser doesn't allow arbitrary strings. I propose this syntax:

Optionally wrap with curly braces: `{{@ocaml foo=[]}[ ... ]}`.
Closing brace can be escaped `\}`, no balancing because it would clash with the other escape.
Not using quotes because it wouldn't be clear where they are allowed, this is consistent with other syntaxes (eg. links), I also thing quotes would be more common in the payload.

Reply:
This PR just does what we [agreed in the dev meeting](https://github.com/ocaml/odoc/wiki/Dev-meeting-2021-02-11). Since the first user of this field will be mdx, do you happen to know whether this is going to be an immediate problem? That is, whether there are any square brackets in any of the current uses of this field in md files?

Reply:
I think mdx only need to be able to read file names, ["relation"](https://github.com/realworldocaml/mdx/blob/master/lib/label.ml#L39), package names, commas, version numbers and pre-defined strings. Escaping isn't needed by Mdx for now.

IMO, we should always think about escaping when writing a language that accepts arbitrary strings unless we are fine with writing "accept any string that doesn't contain '['" in the doc.